Showing enemy stats and other selection fixes

dlougharn opened the pull request:
Implemented selecting enemies and being able to view their stats through radii and action buttons. Certain new features like seeing action damage and range also apply to player units.

https://trello.com/c/VMY7UK7Z/39-show-stats-when-enemy-is-selected

dlougharn committed:
Basic Enemy Select

Enemies can now be selected by the player. GetSelectedCharacter now searches through all player and enemy characters. Added an extra condition in MoveToTile to ensure enemies can't be moved by the player.

dlougharn committed:
Merge branch 'main' into enemybranch

dlougharn committed:
Merge branch 'main' into enemybranch

dlougharn committed:
Enemy Radius

Viewing an enemy's radius now works properly without interferring with the move controller. As a result, move radius will appear for all characters regardless if they can move or not.

dlougharn committed:
Enemy Action Radius + Attack Bugfix

Enemy's action radius can now be viewed. Fixed a bug where melee ranged enemies would not attack because of the logic in GetMostEffectiveAttack.

dlougharn committed:
Basic Damage Forecast UI

Implemented a basic text UI that shows how much range or damage an action has.

saba-kal commented on Vivarium/Assets/Scripts/Characters/CharacterController.cs:
```diff
@@ -46,11 +46,6 @@ protected virtual void VirtualStart()
-        if (!_hasMoved)
-        {
-            ShowMoveRadius();
-        }
```
Is there a reason why this was removed? When a player selects a character, we should show their move radius by default. Removing this code means the player has to select the character, then click the move button.

saba-kal commented on Vivarium/Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -112,6 +112,8 @@ private void SelectAction(Action action)
         _selectedActionRange = StatCalculator.CalculateStat(action, StatType.AttackRange);
         attackViewer.DisplayAction(actionAOE, _selectedActionRange);
+        UIController.Instance.MoveRangeText.text = "R: " + _selectedAction.Range.ToString("N0");
+        UIController.Instance.DamageText.text = "D: " + _selectedAction.BaseDamage.ToString("N0");
```
I would move this logic into the `UIController`. Basically, create a `DisplayActionStats` method and call that here. So, it would look something like this:
```
UIController.Instance.DisplayActionStats(_selectedAction);
```
The reason I'm asking this is because it is generally a good practice to separate unrelated logic. In other words, `PlayerController` should only have to worry about handling player controls while `UIController` should worry about displaying the correct UI.

saba-kal commented on Vivarium/Assets/Scripts/Player/PlayerController.cs:
```diff
@@ -133,6 +135,8 @@ private void DeselectAction()
         _actionIsSelected = false;
         _selectedAction = null;
+        UIController.Instance.MoveRangeText.text = null;
+        UIController.Instance.DamageText.text = null;
```
Similar issue to the one I noted above. I would make a `ClearActionStats` function in `UIController` and call it here.

saba-kal reviewed (changes requested):
Great job! I just have a few small comments.

dlougharn committed:
Player and UI Controller Edits

Made changes in UIController and PlayerController based on Saba's comments in the initial pull.

saba-kal committed:
Re-added showing move radius by default.

saba-kal committed:
Merged in master.
